Mathlib/Analysis/InnerProductSpace/Dual.lean

```diff
@@ -69,6 +70,64 @@ variable {E}
 theorem toDualMap_apply {x y : E} : toDualMap 𝕜 E x y = ⟪x, y⟫ :=
   rfl
 
+section NullSubmodule
+
+open SeparationQuotientAddGroup LinearMap
+
+variable (E)
+
+/-- The null space with respect to the norm. -/
+def nullSubmodule : Submodule 𝕜 E where
+  __ := nullSubgroup
+  smul_mem' c x (hx : ‖x‖ = 0) := show ‖c • x‖ = 0 from
+    le_antisymm (norm_smul_le _ _ |>.trans <| by rw [hx, mul_zero]) (norm_nonneg _)
+
+@[simp]
+lemma mem_nullSubmodule_iff {x : E} : x ∈ nullSubmodule 𝕜 E ↔ ‖x‖ = 0 := Iff.rfl
+
+lemma inner_eq_zero_of_left_mem_nullSubmodule (x y : E) (h : x ∈ nullSubmodule 𝕜 E) :
```

Comment:
This should be stated as
```suggestion
lemma inner_eq_zero_of_left (x y : E) (h : ‖x‖ = 0) :
```
or similar, and can hopefully go in `InnerProducSpace/Basic`.

The existence of `mem_nullSubmodule_iff` as a `@[simp]` lemma means that you should avoid ever writing `∈ nullSubmodule`.